lua/nvim-treesitter/ts_utils.lua

```diff
@@ -145,7 +147,15 @@ function M.update_selection(buf, node)
   end_col = end_col + 1
 
   vim.fn.setpos(".", { buf, start_row, start_col, 0 })
-  vim.fn.nvim_exec("normal! v", false)
+
+  -- Start visual selection in appropriate mode
+  local v_table = {charwise = "v", linewise = "V", blockwise = "<C-v>"}
```

Comment:
you could just use `v_table[selction_mode] or selction_mode` to also allow `<c-v>` in the optional parameter.

Reply:
Do you suggest that possible values of `seleciton_mode` are **only** "v", "V" and "\<C-v\>", or **all** "charwise", "linewise", "blockwise", "v", "V" and "\<C-v\>"? I didn't like the first approach because it introduces cryptic values, and second one because it introduces duplicates.

But this has nothing to do with `nvim-treesitter-textobjects` PR, because there mapping is done from different outputs of Vim's functions to appropriate values of `selection_mode`. And those are different depending on keymapping mode.

Reply:
> all "charwise", "linewise", "blockwise", "v", "V" and "<C-v>"

all of course. first three are for human usage and the rest to pipe vims abbreviation of these modes as keymaps from other functions.

> introduces duplicates

so what? `v_table[selction_mode] or selction_mode` is much shorter than the alternative that would be necessary when you want to update from vim's abbreviation. Selection mode can either be of the three string codes that you suggested or the corresponding key abbreviation used by vim.

Reply:
Sure, no problem. I'll just make sure that "\<C-v\>" can be supplied either as 5 character string or as a single character.